File: tests/fixtures/ts-project/src/order-controller.ts
import { OrderService } from './order-service';
import { PaymentService } from './payment-service';

export class OrderController {
  private orderService: OrderService;

  constructor(orderService: OrderService) {
    this.orderService = orderService;
  }

  async createOrder(request: CreateOrderRequest): Promise<OrderDto> {
    return this.orderService.create(request);
  }

  async getOrder(id: string): Promise<OrderDto> {
    return this.orderService.findById(id);
  }
}

export interface CreateOrderRequest {
  items: string[];
  customerId: string;
}

export interface OrderDto {
  id: string;
  status: string;
}

File: tests/fixtures/python-project/order_controller.py
from .order_service import OrderService


class OrderController:
    def __init__(self, order_service: OrderService):
        self.order_service = order_service

    def create(self, items: list, customer_id: str) -> dict:
        return self.order_service.create_order(items, customer_id)

    def get(self, order_id: str) -> dict:
        return self.order_service.get_order(order_id)
